hhs_oauth_server/request_logging.py

```diff
@@ -1,9 +1,85 @@
 import logging
 import datetime
 import uuid
+import binascii
+import json
+from django.conf import settings
+from apps.fhir.bluebutton.utils import (get_ip_from_request,
+                                        get_user_from_request,
+                                        get_access_token_from_request)
+from apps.accounts.models import get_user_id_salt
+from django.utils.crypto import pbkdf2
+from oauth2_provider.models import AccessToken
+
 
 logger = logging.getLogger('performance.%s' % __name__)
 
+
+class RequestResponseLog(object):
+    """Audit Log message to JSON string
+
+    The JSON log format contians the following fields:
+        - start_time = Unix Epoch format time of the request processed.
+        - end_time = Unix Epoch format time of the response processed.
+        - request_uuid = The UUID identifying the request.
+        - path = The request.path.
+        - response_code = The response status code.
+        - size = Size in bytes of the response.content
+        - location = Location (redirect) for 300,301,302,307 response codes.
+        - user = Login user (or None) or OAuth2 API.
+        - ip_addr = IP address of the request, account for the possibility of being behind a proxy.
+        - access_token_hash = A hash of the access token.
+        - app_name = Application name.
+        - app_id = Application id.
+        - dev_name = Developer user name.
+        - dev_id = Developer user id.
+    """
+
+    request = None
+    response = None
+
+    def __init__(self, req, resp):
+        self.request = req
+        self.response = resp
+
+    def __str__(self):
+        # Create log message dict
+        log_msg = {}
+        log_msg['start_time'] = self.request._logging_start_dt.timestamp()
+        log_msg['end_time'] = datetime.datetime.utcnow().timestamp()
+        log_msg['request_uuid'] = str(self.request._logging_uuid)
```

Comment:
I don't think we need any of these calls to `str` now that we're using `json.dumps`

Reply:
Removing the str() from everything except uuid and user type objects worked OK. For those it will produce an "object of type 'xxx' is not JSON serializable". I removed a few of the str()'s.